Mirror cleanup and enhancements

bfournie opened the pull request:
Various cleanup and changes based on review and discussion

andfasano commented:
Thanks @bfournie for these changes, as they improve considerably the OC_MIRROR feature. I've still pending a couple of non blocking points:
* Automatically changing `MIRROR_IMAGES` when `OC_MIRROR` is set still sounds a little bit weird to me, given that the OC_MIRROR indicates _a way_ to perform the mirroring (in contrast to `oc adm mirror ...` approach)
* When running locally, the `.oc-mirror.log` file is still getting produced in the _current directory_, and not in the working dir

/approve

bfournie commented:
> Thanks @bfournie for these changes, as they improve considerably the OC_MIRROR feature. I've still pending a couple of non blocking points:
>

> * When running locally, the `.oc-mirror.log` file is still getting produced in the _current directory_, and not in the working dir
> 
> /approve



> Thanks @bfournie for these changes, as they improve considerably the OC_MIRROR feature. I've still pending a couple of non blocking points:
> 
> * Automatically changing `MIRROR_IMAGES` when `OC_MIRROR` is set still sounds a little bit weird to me, given that the OC_MIRROR indicates _a way_ to perform the mirroring (in contrast to `oc adm mirror ...` approach)

Changed OC_MIRROR to MIRROR_COMMAND to identify the command used to do the mirroring. MIRROR_IMAGES will be needed to be set

> * When running locally, the `.oc-mirror.log` file is still getting produced in the _current directory_, and not in the working dir
> 

Fixed one spot where oc-mirror was not being run in WORKING_DIR

> /approve

bfournie commented:
Please review, I don't plan on any more changes to this PR but will have a follow-on PR to remove the `OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE` setting in general

bfournie commented:
/retest

lranjbar commented on config_example.sh:
```diff
@@ -559,17 +559,18 @@ set -x
+# "virthost.ostest.test.metalkube.org:LOCAL_REGISTRY_PORT=": { "auth": "<auth string>" },
 #export DOCKER_CONFIG_FILE=$HOME/.docker/config.json
 
 # MIRROR_OLM -
```
MIRROR_OLM isn't supported by the `oc mirror` command yet.

lranjbar reviewed (commented):
We should just call out the gap with `oc mirror` command and MIRROR_OLM. In the future we should refactor our MCE install to use MIRROR_OLM. But this is a follow up story and isn't high priority.

lranjbar commented:
/lgtm
